### PR TITLE
added map tile rendering and MVT info to Maps API doc- WIP

### PR DESCRIPTION
**Pull request by csobier**
Fixes https://github.com/CartoDB/docs/issues/1236

@inigomedina, I know you said to keep this high-level for now so I'm not sure if it needs full examples? Also, I wasn't sure if I am confusing the "Vector rendering" feature with the "MVT" feature, but I think they are one in the same. This is a just rough draft to start, please review.

### CODE REVIEW

File: docs/general_concepts.md [hunk 1]
```diff
@@ -22,6 +22,24 @@ Errors are reported using standard HTTP codes and extended information encoded i
...
+
+**Note:** By default, CARTO uses vector graphics for map rendering. Please [contact us](mailto:support@carto.com) if you need raster rendering enabled as part of your requirements.
+
+### Mapbox Vector Tiles (MVT)
+
+[Mapbox Vector Tiles (MVT)](https://www.mapbox.com/vector-tiles/specification/) are map tiles that store geographic vector data on the client-side. Browser performance is fast since you can pan and zoom without having to query the server.
+
+CARTO uses a Web Graphics Library (WebGL) to process MVT files. This is useful since WebGL's are compatible with most web browsers, include support for multiple client-side mapping engines, and do not require additional information from the server; which makes it more efficient for rendering map tiles.
+
+**Tip:** You can process MVT files with the [`ST_AsMVT` PostGIS function](https://postgis.net/docs/manual-dev/ST_AsMVT.html) with the [Maps API Windshaft renderer](https://github.com/CartoDB/Windshaft/blob/1000x/lib/windshaft/renderers/pg_mvt/renderer.js).
```
Comment (javisantana) [inline docs/general_concepts.md]: be careful, this is not ready yet, databases neet postgis 2.4 and not sure if we should talk about implementation details here

File: docs/general_concepts.md [hunk 2]
```diff
@@ -22,6 +22,24 @@ Errors are reported using standard HTTP codes and extended information encoded i
...
+
+## Map Tile Rendering
+
+Map tiles create the graphical representation of your map in a web browser. The performance rendering of map tiles is dependent on the type of geospatial data model (raster or vector) that you are using.
+
+- **Raster**: Generates map tiles based on a grid of pixels to represent your data. Each cell is a fixed size and contains values for particular map features. On the server-side, each request queries a dataset to retrieve data for each map tile. The grid size of map tiles can often lead to graphic quality issues.
+
+- **Vector**: Generates map tiles based on pre-defined coordinates to represent your data, similar to how basemap image tiles are rendered. On the client-side, map tiles represent real-world geometries of a map. Depending on the coordinates, vertices are used to connect the data and display points, lines, or polygons for the map tiles.
+
+**Note:** By default, CARTO uses vector graphics for map rendering. Please [contact us](mailto:support@carto.com) if you need raster rendering enabled as part of your requirements.
```
Comment (javisantana) [inline docs/general_concepts.md]: be careful, at api level CARTO does not use anything by default. We try to use vector in embeds so please, be careful with wording

File: docs/general_concepts.md [hunk 3]
```diff
@@ -22,6 +22,24 @@ Errors are reported using standard HTTP codes and extended information encoded i
...
+
+### Mapbox Vector Tiles (MVT)
+
+[Mapbox Vector Tiles (MVT)](https://www.mapbox.com/vector-tiles/specification/) are map tiles that store geographic vector data on the client-side. Browser performance is fast since you can pan and zoom without having to query the server.
+
+CARTO uses a Web Graphics Library (WebGL) to process MVT files. This is useful since WebGL's are compatible with most web browsers, include support for multiple client-side mapping engines, and do not require additional information from the server; which makes it more efficient for rendering map tiles.
```
Comment (javisantana) [inline docs/general_concepts.md]: again, we use webgl based renderer in embeds but the user could use any implementation (webgl based or not)

**Comment by javisantana** [comment]
I miss the way to actually get the mvt tiles for developers

File: docs/general_concepts.md (hunk 2 above)
Comment (csobier) [reply docs/general_concepts.md]: 👍  Removing this note

**Comment by inigomedina** [comment]
Do we have any content for developers? I know the [Mapnik Vector Tiles doc](https://docs.google.com/document/d/1AtlwQ1zxTfHs30K95eEBIfa-vUNYUZOYP9rJ_iseg3s/edit#heading=h.9erwjv4dhent), where there are context and some API examples.

How do you find that @javisantana?

**Comment by csobier** [comment]
I found this online too https://carto.com/blog/inside/MVT-mapnik-vs-postgis/

**Comment by javisantana** [comment]
We have a full section in maps api for developers who want to use our API: 

https://github.com/CartoDB/Windshaft-cartodb/blob/master/docs/anonymous_maps.md#retrieve-resources-from-the-layergroup

So we should add:
- how to get mvt tiles
- extra information about MVT like how to change buffer, default buffer size, default resolution and so on
- limitations like number of features per tile, simplification issues, torque, merging layers and so on
- examples on how to fetch mvt (using curl for example)

cc @rochoa

**Pull request merged**